### ISSUE TITLE
How to hide "choose your look" on non-gnome shell installers?

### ISSUE THREAD

**Issue by fossfreedom** [canonical/ubuntu-desktop-installer#1533]
The dark / light theme option is specific to gnome-shell.

For Ubuntu Budgie I don't think we can configure this page via ubuntu-flavor-installer - am I mistaken here?

Is there a way that this page can only be displayed for gnome shell only?

Or alternatively can this page be added to be configurable for flavors?

At the moment selecting light breaks the budgie desktop visually since yaru has little support for budgie desktop.

**Comment by jpnurmi** [comment]
You're right that in the longer term we probably need a way to control which pages are shown for flavors but we could start with a simple fix to make the installer respect the system theme.

The installer clearly shouldn't blindly switch the theme between "Yaru" and "Yaru-dark". It's been like this since the early days of the new installer and simply wasn't paid attention to because it didn't cause issues until now. :)

We could easily make the installer query the current system theme and then add or remove the "-dark" suffix, for example. That way it would switch between "Pocillo" and "Pocillo-dark". Would it do the right thing on Budgie?

**Comment by fossfreedom** [comment]
Hi. Adding a suffix to the system theme would make sense from a  live session short-term point of view.

  I don't have any insight though how that choice carries forward after the installation for the new login that is created and for any other user account that is subsequently created.

**Comment by jpnurmi** [comment]
> I don't have any insight though how that choice carries forward after the installation for the new login that is created and for any other user account that is subsequently created.

This is handled by the post-installation scripts:
https://github.com/UbuntuBudgie/ubuntu-budgie-installer/blob/main/snap/local/postinst.d/10_override_desktop_settings

It might be good to review to scripts to see if there's anything you want/need to customize for Budgie:
https://github.com/UbuntuBudgie/ubuntu-budgie-installer/tree/main/snap/local/postinst.d

**Comment by jpnurmi** [comment]
The next round of updates should help:
- https://github.com/UbuntuBudgie/ubuntu-budgie-installer/pull/4

**Comment by jpnurmi** [comment]
There's currently no easy way to hide the screen but the default desktop settings service no longer hardcodes Yaru _and_ there's a way to pre-register a custom implementation of the desktop service:

```dart
abstract class DesktopService {
  Future<void> inhibit();
  Future<void> setTheme(Brightness brightness);
  Future<void> close();
}
```
For example:
```dart
void main() {
  registerService<DesktopService>(MyDesktopService.new);
  runInstallerApp(...);
}
```
